packages/components/src/text-input/hooks/use-text-input.js

```diff
@@ -0,0 +1,222 @@
+	isTypeNumeric,
+} ) => {
+	const canScroll = isTypeNumeric && isFocused;
+	const dragHandlers = dragHandlersRef.current;
+
+	useEffect( () => {
+		const handleOnWheel = () => {};
+		if ( inputRef.current.addEventListener ) {
+			inputRef.current.addEventListener( 'wheel', handleOnWheel, {
+				passive: false,
+			} );
+		}
+	}, [ inputRef, canScroll ] );
+
+	const handleOnClick = useCallback( () => {
+		inputRef.current.focus();
+	}, [ inputRef ] );
+
+	const handleOnTouchStart = useCallback( () => {
+		inputRef.current.focus();
+	}, [ inputRef ] );
+
+	return {
+		...dragHandlers,
+		onClick: handleOnClick,
+		onTouchStart: handleOnTouchStart,
+	};
+};
+
+/**
+ * @typedef ReturnProps
+ * @property {(int: number) => void} [increment] Increment text input number value callback.
+ * @property {(int: number) => void} [decrement] Decrement text input number value callback.
+ * @property {import('react').MutableRefObject<null | HTMLElement>} [dragHandlersRef] Renders an error.
+ * @property {import('react').RefObject<HTMLInputElement>} [inputRef] Whether the field is disabled.
+ * @property {boolean} [isTypeNumeric] Whether the type is numeric.
+ * @property {boolean} [isInputTypeNumeric] Whether the input type is numeric.
+ * @property {string} [className] Classes
+ * @property {Object} [inputProps] Input props
+ */
+
+/** @typedef {import('../types').Props & ReturnProps} ReturnPropsObject */
+
+/**
+ *
+ * @param  {import('../../ui/context').PolymorphicComponentProps<import('../types').Props, 'input'>} props
+ * @return {ReturnPropsObject} props
+ */
+export function useTextInput( props ) {
+	const {
+		align,
+		arrows = true,
+		className,
+		defaultValue = '',
+		disabled,
+		dragAxis,
+		error = false,
+		format,
+		gap = 2.5,
+		id: idProp,
+		incrementFromNonNumericValue = false,
+		isCommitOnBlurOrEnter = false,
+		isFocused: isFocusedProp,
+		isInline = false,
+		isResizable = false,
+		isShiftStepEnabled = true,
+		justify,
+		max,
+		min,
+		multiline = false,
+		prefix,
+		shiftStep = 10,
+		size = 'medium',
+		step,
+		suffix,
+		type = 'text',
+		validate,
+		value: valueProp,
+		...otherProps
+	} = useContextSystem( props, 'TextInput' );
+
+	const id = useFormGroupContextId( idProp );
+
+	const {
+		decrement,
+		dragHandlersRef,
+		increment,
+		inputRef,
+		isFocused,
+		isInputTypeNumeric,
+		isTypeNumeric,
+		scrollHandlers,
+		value,
+		...textInputState
+	} = useTextInputState( {
+		...otherProps,
+		format,
+		defaultValue,
+		incrementFromNonNumericValue,
+		isCommitOnBlurOrEnter,
+		isFocused: isFocusedProp,
+		isShiftStepEnabled,
+		max,
+		min,
+		shiftStep,
+		step,
+		value: valueProp,
+		validate,
+		type,
+	} );
+
+	const rootEventHandlers = useRootEventHandlers( {
+		inputRef,
+		dragHandlersRef,
+		isTypeNumeric,
+		isFocused,
+	} );
+
+	const baseFieldProps = useBaseField( {
+		align,
+		disabled,
+		// @ts-ignore Looks like we need to extend the Props @typedef here? packages/components/src/base-field/hook.js
```

Comment:
`useBaseField` was changed from the original G2 implementation, we can just remove/rename (in the case of error it was renamed to `hasError` I believe) the props it no longer accepts.

Reply:
Thanks for the guidance here. I removed the props that `useBaseField` no longer specifies, and renamed `hasError`.

```console
TS2345: Argument of type '{ disabled: boolean | undefined; hasError: boolean; isInline: boolean; }' is not assignable to parameter of type 'PolymorphicComponentProps<Props, "div">'.   

Property 'children' is missing in type '{ disabled: boolean | undefined; hasError: boolean; isInline: boolean; }' but required in type 'FlexProps'.

```

I've ignored the line for now as I'm not sure how to tackle that.